docs/basic-reader-and-writer.md

```diff
@@ -366,4 +366,42 @@ altered and the set of Streams being read has therefore been changed.
 So that's it.  The simple HelloWorldReader loops, reading Events from a Stream
 until there are no more Events, and then the application terminates.
 
- 
+## ReaderGroup Notifications
+
+ReaderGroup api supports different types of notifications. The supported
+notifications are
+
+1. SegmentEvent Notification
+
+A segment event is triggered when the total number of segments managed by the
+ReaderGroup changes. During a scale operation segments can be split into
+multiple or merged into some other segment causing the total number of segments
+to change. The total number of segments can also change when configuration
+of the reader group is changed, for example modify the configuration of a reader
+group to add/remove a stream.
+
+The method for subscribing to SegmentEvent notifications is shown below
+```
+@Cleanup
+ReaderGroupManager groupManager = new ReaderGroupManagerImpl(SCOPE, controller, clientFactory,
+        connectionFactory);
+ReaderGroup readerGroup = groupManager.createReaderGroup(GROUP_NAME, ReaderGroupConfig
+        .builder().build(), Collections.singleton(STREAM));
+
+readerGroup.getSegmentEventNotifier(executor).registerListener(segmentEvent -> {
+       int numOfReaders = segmentEvent.getNumOfReaders();
+       int segments = segmentEvent.getNumOfSegments();
+       if (numOfReaders < segments) {
+          //Scale up number of readers based on application capacity
+       } else {
+         //More readers available time to shut down some
+       }
+});
+
+```
+The application can register a listener to be notified of SegmentEvents using the registerListener api.
```

Comment:
If `SegmentEvent` refers to objects of that type, then it should be one word and code formatted with ``. Otherwise, if it is about the concept, then we should use segment events instead.

Reply:
ok. updated it.